src/components/message/index.tsx

```diff
@@ -59,85 +58,82 @@ interface Properties extends MessageModel {
-  isAttachmentPreviewOpen: boolean;
-  attachmentPreview: {
-    name: string;
-    url: string;
-    type: string;
-    mimetype?: string;
-  } | null;
-  isReactionPickerOpen: boolean;
-}
-
-export class Message extends React.Component<Properties, State> {
-  state = {
-    isEditing: false,
-    isFullWidth: false,
-    isMessageMenuOpen: false,
-    isDropdownMenuOpen: false,
-    menuX: 0,
-    menuY: 0,
-    isImageLoaded: false,
-    isAttachmentPreviewOpen: false,
-    attachmentPreview: null,
-    isReactionPickerOpen: false,
-  } as State;
-
-  wrapperRef = React.createRef<HTMLDivElement>();
-
-  handleContextMenu = (event) => {
-    if (typeof window !== 'undefined' && window.getSelection) {
-      const selectedText = window.getSelection().toString();
-      if (selectedText.length > 0) {
-        return;
-      }
-    }
-
-    if (event.button === 2) {
-      event.preventDefault();
-      event.stopPropagation();
-      const { pageX, pageY } = event;
-      this.setState({
-        isDropdownMenuOpen: true,
-        isMessageMenuOpen: false,
-        menuX: pageX,
-        menuY: pageY,
-      });
-    }
-  };
-
-  openAttachmentPreview = async (attachment): Promise<void> => {
-    this.setState({ isAttachmentPreviewOpen: true, attachmentPreview: attachment });
-  };
-
-  closeAttachmentPreview = () => {
-    this.setState({ isAttachmentPreviewOpen: false, attachmentPreview: null });
-  };
-
-  downloadAttachment = async () => {
-    if (!this.state.attachmentPreview) return;
-    await download(this.state.attachmentPreview.url);
-  };
-
-  downloadImage = (media) => () => {
+export const Message: React.FC<Properties> = ({
+  className,
+  onImageClick,
+  onDelete,
+  onEdit,
+  onInfo,
+  onReply,
+  isOwner,
+  messageId,
+  parentSenderIsCurrentUser,
+  parentSenderFirstName,
+  parentSenderLastName,
+  parentMessageMediaUrl,
+  parentMessageMediaName,
+  getUsersForMentions,
+  showSenderAvatar,
+  showTimestamp,
+  showAuthorName,
+  isHidden,
+  onHiddenMessageInfoClick,
+  loadAttachmentDetails,
+  sendEmojiReaction,
+  onReportUser,
+  messagesFetchStatus,
+  media,
+  message,
+  parentMessageText,
+  parentMessageMedia,
+  parentMessageId,
+  isAdmin,
+  createdAt,
+  updatedAt,
+  sender,
+  mentionedUsers,
+  hidePreview,
+  preview,
+  admin,
+  optimisticId,
+  rootMessageId,
+  sendStatus,
+  reactions,
+}) => {
+  const [isEditing, setIsEditing] = useState(false);
+  const [isMessageMenuOpen, setIsMessageMenuOpen] = useState(false);
+  const [isAttachmentPreviewOpen, setIsAttachmentPreviewOpen] = useState(false);
+  const [attachmentPreview, setAttachmentPreview] = useState(null);
+  const [isReactionPickerOpen, setIsReactionPickerOpen] = useState(false);
+  const [isDropdownMenuOpen, setIsDropdownMenuOpen] = useState(false);
+
+  const isMenuTriggerAlwaysVisible = sendStatus === MessageSendStatus.FAILED;
+
+  const senderAvatarUrl = useMemo(() => {
+    return getProvider().getSourceUrl(sender.profileImage);
```

Comment:
@ratik21 you mentioned that we shouldn't be getting avatars this way. Can you point me to another usage where we're getting the avatar from the correct location?

Reply:
as discussed on call, i think we can just remove this.